### PR TITLE
Reactive labels

### LINKED ISSUE #7: Handle $: reactive label
image

Reactive label should display as keyword and don't prompt inspections. Implicit `let` keyword insertion in also needed.

### PR DESCRIPTION
**Pull request by tomblachut**
Resolves #7

### CODE REVIEW

**Comment by anstarovoyt** [comment]
is it really necessary to introduce a new PSI element for labels? Maybe it is better just suppress the error using the corresponding EP?

File: src/main/java/dev/blachut/svelte/lang/parsing/js/SvelteJSParser.kt [hunk 1]
```diff
@@ -16,6 +21,40 @@ class SvelteJSParser(builder: PsiBuilder) : ES6Parser<ES6ExpressionParser<*>, ES
...
+                // Treat reactive statements as normal labelled statements
+                val reactiveDeclaration = reactiveLabel && isIdentifierToken(builder.lookAhead(2))
+                if (!reactiveDeclaration) return false
+
+                if (isAfterModifierKeyword()) {
+                    builder.error("reactive declarations can't be exported")
+                }
+
+                builder.advanceLexer()
+                builder.advanceLexer()
```
Comment (eriknyk) [inline src/main/java/dev/blachut/svelte/lang/parsing/js/SvelteJSParser.kt:41]: Why calling `builder.advanceLexer()` twice?

Comment (tomblachut) [reply src/main/java/dev/blachut/svelte/lang/parsing/js/SvelteJSParser.kt:41]: because `$` and `:` are two separate tokens

**Comment by shirotech** [comment]
Screen Shot 2020-08-09 at 3 22 13 pm

Probably I should report it here, when importing a store and using it with $, doesn't seem to recognize it as such.

**Comment by tomblachut** [comment]
> is it really necessary to introduce a new PSI element for labels? Maybe it is better just suppress the error using the corresponding EP?

This is not new PSI element, PR changes the parser so it treats label as normal var statement.
I did it like that so resolving works with JSReferenceExpressionResolver without writing complicated rules about walking up the tree etc.

Downside is that there are inspections that require actual let and const keywords, IDK if this is correct approach.

Alternatively, there was great work performed by @unlocomqx in #43, it contains some edge case bugs though.

(Woah, that year certainly flew by)

**Comment by anstarovoyt** [comment]
>Downside is that there are inspections that require actual let and const keywords, IDK if this is correct approach.

We have fields and parameters that extend almost the same interfaces and don't have any keywords so in general, I think such solution is ok.

**Comment by tomblachut** [comment]
#163 contains alternative approach

**Pull request closed without merging**